OutOfMemoryError when NASA validate v3.5.2 is executed through a library for a batch of products

### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When NASA validate v3.5.2 is executed through a JAR library for a batch of products in a Java v17 project, I noticed that the heap memory is not released in each product validate iteration so when the library validated some products an OutOfMemoryError exception is raised.

```
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.apache.catalina.core.StandardServer.startPeriodicLifecycleEvent(StandardServer.java:937)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.OutOfMemoryError: Java heap space
```

I've seen a similar issue: https://github.com/NASA-PDS/validate/issues/826 but it seems that this is a bit different.

### 🕵️ Expected behavior

I'd expect that the memory was released in each product validate iteration or at least that the memory was released before reach the heap size to avoid the out of memory exception.

### 📜 To Reproduce

0. Create a Java 17 project that uses NASA validate v3.5.2 as a jar project library.
1. Create a ValidateLauncher object and a String array intended to be use as arguments for ValidateLauncher object.
```
final ValidateLauncher launcher = new ValidateLauncher();
String[] arguments = {
                    "-C", "pds4-validate-catalog.xml",
                    "-e", "xml",
                    "-E", String.valueOf(Long.MAX_VALUE),
                    "-s", "json",
                    "--add-context-products", "local_context_products.json",
                    "-t", "cam_raw_sc_cam3_image_20210812t041402_61_f__t0005.xml",
                    "-r", "/tmp/PDS4_VALIDATION_REPORTS/pds_validate_report_bc_mtm_mcam_1724407870771_b296d73d-9536-4ac3-baee-9b40faf4fab3.json"
            };
```
[pds4-validate-catalog.zip](https://github.com/user-attachments/files/16727482/pds4-validate-catalog.zip)
[local_context_products.json](https://github.com/user-attachments/files/16727439/local_context_products.json)
[cam_raw_sc_cam3_20210812t041402_61_f__t0005.zip](https://github.com/user-attachments/files/16727542/cam_raw_sc_cam3_20210812t041402_61_f__t0005.zip)
2. Call `launcher.processMain(arguments);` and release the object:
```
            launcher.flushValidators();
            CrossLabelFileAreaReferenceChecker.reset();
```
3. Repeat 1 and 2 several times and check the memory heap usage (you can use the VisualVM tool to analyze it).

### 🖥 Environment Info

- Version of this software v3.5.2
- Operating System: Ubuntu 20.04.6 LTS
- Kernel: Linux 5.4.0-193-generic
- Architecture: x86-64
- Java version: 17

### 📚 Version of Software Used

Version 3.5.2

### 🩺 Test Data / Additional context

VisualVM v1.4.3 was used to analyze the heap memory and as we can see, the heap memory is not released:

![heapMemory](https://github.com/user-attachments/assets/bc1d4832-5ecc-4653-b0f1-3f448be2d606)

Finally, an out of memory exception happen.

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -135,7 +135,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-site-plugin</artifactId>
-        <version>4.0.0-M16</version>
+        <version>4.0.0-M13</version>
         <configuration>
           <skip>false</skip>  <!-- This is `true` in parent pom -->
           <skipDeploy>false</skipDeploy>  <!-- This is `true` in parent pom -->

--- a/src/site/markdown/operate/index.md
+++ b/src/site/markdown/operate/index.md
@@ -26,7 +26,8 @@ This document describes how to operate the Validate Tool. The following topics c
         - [pds4.collection](#pds4collection)
         - [pds3.volume](#pds3volume)
     - [Referential Integrity Checking](#referential-integrity-checking)
-    - [Advanced Bundle Validation](#advanced-bundle-validation)
+    - [Improve Performance](#improve-performance)
+        - [Batching](#batching)   
         - [Referential Integrity Checking Only](#referential-integrity-checking-only)
         - [Specify Additional File Paths for Accumulating Bundles](#specify-additional-file-paths-for-accumulating-bundles)
         - [Example of using multiple-flags to improve performance](#example-of-using-multiple-flags-to-improve-performance)
@@ -385,12 +386,34 @@ In the case that 2 targets are specified on the command line, the tool will trea
 ```
 The tool will perform referential integrity checking within the products located in the _$\{HOME\}/bundle1_ target, then will perform referential integrity checking within the products located in the _$\{HOME\}/bundle2_ target. In other words, the tool will not cross over to the _$\{HOME\}/bundle2_ directory to find LID/LIDVID references of a product in the _$\{HOME\}/bundle1_ directory.
 
-### Advanced Bundle Validation
+### Improve Performance
+
+Validate is a very memory intensive applications that requires a significant amount of resources and time in order to complete validation on large data sets.
 
 There are a few ways you can enhance the way validate executes PDS4 Bundle validation using several options. There are several flags added to validate so you can split out various parts
 and run multiple Validate executions to speed up the processing. However, to ensure proper validity of the archive, several of these flags should
 
 only be used during development or staging of data prior to release.
+
+#### Batching
+
+For both memory and performance improvements, it is highly recommended that you batch large bundles, both in terms of data volume and number of products, into batches.
+
+You can batch either by directory or by using target manifests.
+
+```
+# Point at several targets with different validate runs for parallelizing content validation
+validate --rule pds4.label --target my_bundle/document/*.xml
+validate --rule pds4.label --target my_bundle/release1/*.xml
+validate --rule pds4.label --target my_bundle/release2/*.xml
+validate --rule pds4.label --target my_bundle/release3/*.xml
+
+# Or you could do the same thing with target manifests
+validate --rule pds4.label --target-manifest documents.txt
+validate --rule pds4.label --target-manifest release1_data.txt
+validate --rule pds4.label --target-manifest release2_data.txt
+validate --rule pds4.label --target-manifest release3_data.txt
+```
 
 #### Referential Integrity Checking Only
 

--- a/src/site/xdoc/operate/errors.xml.vm
+++ b/src/site/xdoc/operate/errors.xml.vm
@@ -80,6 +80,12 @@ Exception in thread "main" java.lang.OutOfMemoryError: GC overhead limit exceede
       <source>
 "%JAVA_HOME%"\bin\java -Xms4096m -Xmx8192m -jar "%VALIDATE_JAR%" %*
       </source> 
+      
+      <p><b>Still not working?</b> 
+      </p>
+      
+      <p>See Operate documentation on how to improve performance, including ensuring your execution is sufficiently batched.
+      </p>
 
     </section>
       

